openpype/tools/sceneinventory/switch_dialog.py

```diff
@@ -1294,17 +1291,32 @@ def _trigger_switch(self, loader=None):
                         repre_doc = _repres.get(container_repre_name)
 
             if not repre_doc:
-                version_by_name = version_docs_by_parent_and_name[subset_id]
+                version_by_name = version_docs_by_parent_id_and_name[subset_id]
+
+                # If asset or subset are selected for switching, we use latest
+                # version else we try to keep the current container version.
                 if selected_asset or selected_subset:
-                    version_doc = last_version_docs_by_parent_id[subset_id]
+                    version_name = max(version_by_name)
                 else:
-                    version_doc = version_by_name[container_version_name]
+                    version_name = container_version_name
+
+                version_doc = version_by_name[version_name]
                 version_id = version_doc["_id"]
                 repres_by_name = repre_docs_by_parent_id_by_name[version_id]
+
                 if selected_representation:
-                    repre_doc = repres_by_name[selected_representation]
+                    repres_name = selected_representation
                 else:
-                    repre_doc = repres_by_name[container_repre_name]
+                    repres_name = container_repre_name
+
+                # If selected representation is not available with container
+                # version, we use latest version.
+                if repres_name not in repres_by_name:
+                    repres_by_name = repre_docs_by_parent_id_by_name[
+                        version_by_name[max(version_by_name)]["_id"]
+                    ]
```

Comment:
The comment hints to me that this should be in the `if selected_representation:` branch?

Also I feel like this logic here is going to be confusing as to when this occurs. Should this at least log a debug or warning about this edge case?

Reply:
Ok I revert this change. This shouldn't be a big problem right now. I have to do some crash testing to be sure of this feature.